paddle/fluid/framework/details/build_strategy.h

```diff
@@ -112,6 +112,11 @@ struct BuildStrategy {
   bool cache_expected_kernel_{true};
   std::unordered_set<std::string> mkldnn_enabled_op_types_;
 
+  size_t nccl_comm_num_{1};
+  bool use_hierarchical_allreduce_{false};
+  size_t hierarchical_allreduce_inter_nranks_{0};
+  size_t hierarchical_allreduce_exter_nranks_{0};
```

Comment:
What's the meaning of `hierarchical_allreduce_inter_nranks_ ` and `hierarchical_allreduce_exter_nranks_ `?

Reply:
![image](https://user-images.githubusercontent.com/10721757/57992255-bd16d380-7ae5-11e9-9211-2f39ec3f16f9.png)

Reply:
Internal(in a node) ranks and external ranks(between nodes).

Reply:
Please add doc in code.

Reply:
Done.

Reply:
You should cite the reference of original image. https://arxiv.org/pdf/1807.11205.pdf